core/src/com/unciv/models/ruleset/Event.kt

```diff
@@ -6,25 +6,40 @@ import com.unciv.models.ruleset.unique.StateForConditionals
 import com.unciv.models.ruleset.unique.Unique
 import com.unciv.models.ruleset.unique.UniqueTriggerActivation
 import com.unciv.models.stats.INamed
+import com.unciv.ui.components.input.KeyCharAndCode
+import com.unciv.ui.screens.civilopediascreen.FormattedLine
+import com.unciv.ui.screens.civilopediascreen.ICivilopediaText
 
 
-
-class Event : INamed {
+class Event : INamed, ICivilopediaText {
 
     override var name = ""
     var text = ""
+    override var civilopediaText = listOf<FormattedLine>()
+    override fun makeLink() = "Event/$name"
+
     // todo: add unrepeatable events
 
     var choices = ArrayList<EventChoice>()
+
+    fun getMatchingChoices(stateForConditionals: StateForConditionals) =
+        choices.filter { it.matchesConditions(stateForConditionals) }.ifEmpty { null }
```

Comment:
ifEmpty{ null } is an interesting choice, I see how that fits later, but it makes the return value of this function a tad unclear
Oh well, probably anyone who'll be looking at this will be using some IDE with type hinting anyway